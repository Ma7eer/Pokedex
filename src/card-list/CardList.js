import React from 'react';
import Card from '../card/Card';

const CardList = (props) => {
  return (
    <div className="d-flex flex-wrap mt-4 align-items-center">
      {props.data.map((card, index) => <Card img={card.img} name={card.name} id={card.id} key={index}/>)}
    </div>
  )
}

export default CardList