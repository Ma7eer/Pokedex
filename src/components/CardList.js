import React from 'react';
import Card from './Card.js';

const CardList = (props) => {
  return (
    <div className="d-flex flex-wrap mt-4 align-items-center justify-content-center my-card">
      {props.data.map((card, index) => <Card {...card} key={index} />)}
    </div>
  )
}

export default CardList