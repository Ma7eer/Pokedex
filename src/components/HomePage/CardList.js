import React from 'react';
import Card from './Card.js';
import PropTypes from 'prop-types';

const CardList = (props) => {
  const { data } = props;
  return (
    <div className="d-flex flex-wrap mt-4 align-items-center justify-content-center my-card">
      {data.map((card, index) => <Card {...card} key={index} />)}
    </div>
  )
}

CardList.propTypes = {
  data: PropTypes.array
}

export default CardList