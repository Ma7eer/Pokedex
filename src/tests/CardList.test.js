import React from 'react';
import renderer from 'react-test-renderer';

import CardList from '../components/Home/CardList';

// running some snapshot tests on out CardList Component
describe('CardList Component', () => {
  it('Should fail if no data is fed into it', () => {
    const tree = renderer.create(<CardList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe('CardList Component', () => {
  it('Should pass if data is fed into it', () => {
    const tree = renderer.create(<CardList
    data={
      [{
      spriteImgUrl: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
          id: "1",
          name: "bulbasaur",
          type: "poison"
        }]
      }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})