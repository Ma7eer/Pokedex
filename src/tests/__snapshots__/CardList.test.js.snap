// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component Should pass if data is fed into it 1`] = `
<div
  className="d-flex flex-wrap mt-4 align-items-center justify-content-center my-card"
>
  <div
    className="bg-dark p-4"
    style={
      Object {
        "margin": "1em",
      }
    }
  >
    <img
      alt=""
      className="hvr-bob"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
      width="180"
    />
    <div
      style={
        Object {
          "display": "inline-block",
          "marginLeft": 10,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": "1.25em",
            "fontWeight": "bold",
          }
        }
      >
        bulbasaur
      </div>
      <div>
        Pokedex ID: 
        1
      </div>
      <div
        className="rounded d-flex justify-content-center mt-2"
        style={
          Object {
            "background": "#a040a0",
          }
        }
      >
        Type: 
        poison
      </div>
      <button
        className="btn btn-primary mt-3"
        onClick={[Function]}
        type="button"
      >
        Add to list
      </button>
    </div>
  </div>
</div>
`;
